students/jayjohnson/lesson02/Grid_Printer.py

```diff
@@ -0,0 +1,151 @@
+|               |               |
+|               |               |
+|               |               |
+|               |               |
+|               |               |
+|               |               |
+|               |               |
++ - - - - - - - + - - - - - - - +
+This problem is underspecified. Do something reasonable.
+'''
+
+def print_grid(n):
+    grid_count_upper = math.floor(n/2)
+    grid_count_lower = math.floor(n/2)
+    print('+ ' + '- '*math.floor(n/2) + '+ ' + '- '*math.floor(n/2) + '+'  )
+    for i in range( grid_count_upper ):
+        print('| ' + '  '*math.floor(n/2) + '| ' + '  '*math.floor(n/2) + '|'  )
+    print('+ ' + '- '*math.floor(n/2) + '+ ' + '- '*math.floor(n/2) + '+'  )
+    for i in range( grid_count_lower):
+        print('| ' + '  '*math.floor(n/2) + '| ' + '  '*math.floor(n/2) + '|'  )
+        grid_count_lower = grid_count_lower - 1
+    print('+ ' + '- '*math.floor(n/2) + '+ ' + '- '*math.floor(n/2) + '+'  )
+
+print_grid(3)
+print_grid(15)
+
+
+'''
+Write a function that draws a similar grid with a specified number of rows and columns, and with each cell a given size.
+
+For example, print_grid2(3,4) results in:
+
++ - - - - + - - - - + - - - - +
+|         |         |         |
+|         |         |         |
+|         |         |         |
+|         |         |         |
++ - - - - + - - - - + - - - - +
+|         |         |         |
+|         |         |         |
+|         |         |         |
+|         |         |         |
++ - - - - + - - - - + - - - - +
+|         |         |         |
+|         |         |         |
+|         |         |         |
+|         |         |         |
++ - - - - + - - - - + - - - - +
+(three rows, three columns, and each grid cell four “units” in size)
+
+What to do about rounding? – you decide.
+
+Another example: print_grid2(5,3):
++ - - - + - - - + - - - + - - - + - - - +
+|       |       |       |       |       |
+|       |       |       |       |       |
+|       |       |       |       |       |
++ - - - + - - - + - - - + - - - + - - - +
+|       |       |       |       |       |
+|       |       |       |       |       |
+|       |       |       |       |       |
++ - - - + - - - + - - - + - - - + - - - +
+|       |       |       |       |       |
+|       |       |       |       |       |
+|       |       |       |       |       |
++ - - - + - - - + - - - + - - - + - - - +
+|       |       |       |       |       |
+|       |       |       |       |       |
+|       |       |       |       |       |
++ - - - + - - - + - - - + - - - + - - - +
+|       |       |       |       |       |
+|       |       |       |       |       |
+|       |       |       |       |       |
++ - - - + - - - + - - - + - - - + - - - +
+'''
+
+def print_grid2(c,n):
+    print('+' + " - "*n + '+')
+    for i in range( n ):
+        print('|' + "   "*n + '|')
+    print('+' + " - "*n + '+')
+    #cant figure out how to print multiple boxes
```

Comment:
Check out the solutions folder and see if that makes sense to you. Let me know if you need more of an explanation. I find it helpful to copy code into my editor, and put in a ton of print() calls to see how things change as the program progresses. Also, commenting out code can help.